openedx/core/djangoapps/user_api/accounts/serializers.py

```diff
@@ -63,6 +68,16 @@ def to_representation(self, user):
         :return: Dict serialized account
         """
         profile = user.profile
+        badges = settings.FEATURES.get('ENABLE_OPENBADGES') or False
+        if 'badges' in self.excluded_fields:
+            # Avoid doing this potentially expensive query-- it will be filtered out anyway.
```

Comment:
I strongly suggest we change this field to a boolean: 'has_accomplishments' for example.
This will allow the Profile UI to know whether to display the "accomplishments" tab.
And then have the UI call the badges API in a separate AJAX request.

This is exactly how we were planning to do it for mobile: boolean in Profile API, with paginated call to Badges API.
If you can do this, it would solve both Mobile's and Web's requirements, I believe.